ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/ExpressionTypeManager.java

```diff
@@ -143,9 +192,8 @@ public Void visitArithmeticUnary(
     public Void visitLambdaExpression(
         final LambdaFunctionCall node, final ExpressionTypeContext context
     ) {
+      context.mapInputTypes(node.getArguments());
```

Comment:
I think we should throw an exception if there aren't an equal number of arguments from the lambda function call and the number of unmapped input types already in the context.

For example:
What if the user passes in MAP_REDUCE(map, state, (x,y) => ...)

We might be handling this already in `visitFunctionCall` when we get the Function from `UdfIndex` but we can be more specific about that scenario

Reply:
Yeah, I think that makes sense. This definitely relies on getting the number in inputs right from the `ExpressionTypeManager` which I think is somewhat of an open question if we're avoiding hard coding function names